docs/api_specs/includes/watcher_api_specs.md

```diff
@@ -0,0 +1,314 @@
+        "proof": "CEDB8B31D1E4CB72EC267A8B27C42C4D9982C3F3950D88003F44B3A797202D848025356282CA1C28CBD51FBF7D8E9187AA85F628D054B2C2233AA83BCAEF1F0EB4C11951957C6F8F642C4AF61CD6B24640FEC6DC7FC607EE8206A99E92410D3021DDB9A356815C3FAC1026B6DEC5DF3124AFBADB485C9BA5A3E3398A04B7BA85E58769B32A1BEAF1EA27375A44095A0D1FB664CE2DD358E7FCBFB78C26A193440EB01EBFC9ED27500CD4DFC979272D1F0913CC9F66540D7E8005811109E1CF2D887C22BD8750D34016AC3C66B5FF102DACDD73F6B014E710B51E8022AF9A1968FFD70157E48063FC33C97A050F7F640233BF646CC98D9524C6B92BCF3AB56F839867CC5F7F196B93BAE1E27E6320742445D290F2263827498B54FEC539F756AFCEFAD4E508C098B9A7E1D8FEB19955FB02BA9675585078710969D3440F5054E0F9DC3E7FE016E050EFF260334F18A5D4FE391D82092319F5964F2E2EB7C1C3A5F8B13A49E282F609C317A833FB8D976D11517C571D1221A265D25AF778ECF8923490C6CEEB450AECDC82E28293031D10C7D73BF85E57BF041A97360AA2C5D99CC1DF82D9C4B87413EAE2EF048F94B4D3554CEA73D92B0F7AF96E0271C691E2BB5C67ADD7C6CAF302256ADEDF7AB114DA0ACFE870D449A3A489F781D659E8BECCDA7BCE9F4E8618B6BD2F4132CE798CDC7A60E7E1460A7299E3C6342A579626D2"
+    }
+}
+```
+
+Gets exit data for a given utxo
+
+### HTTP Request
+
+`POST /utxo.get_exit_data`
+
+### JSON Body
+
+Key | Type | Description
+--------- | ------- | -----------
+utxo_pos | Integer | Utxo position (encoded as single integer, the way contract represents them)
+
+
+
+## Transaction - Submit
+
+```shell
+http POST /transaction.submit transaction=f8d083015ba98080808080940000...
+```
+
+```elixir
+// TODO
+```
+
+```javascript
+// TODO
+```
+
+> The above command returns JSON document:
+<aside class="warning">Is it OK? Following is planned child chain response for corresponding endpoint</aside> 
+
+```json
+{
+      "version": "1",
+      "success": true,
+      "data": { 
+          "blknum": 123000,
+          "txindex": 111,
+          "txhash": "bdf562c24ace032176e27621073df58ce1c6f65de3b5932343b70ba03c72132d"
+      }
+}
+```
+
+Watcher passes signed transaction to the child chain only if it's secure (better explaination needed)
+
+### HTTP Request
+
+`POST /transaction.submit`
+
+### JSON Body
+
+Key | Type | Description
+--------- | ------- | -----------
+transaction | Hex encoded string | Signed transaction RLP-encoded to bytes and HEX-encoded to string
+
+
+
+
+## Status
+
+```shell
+http POST /status
+```
+
+```elixir
+// TODO
+```
+
+```javascript
+// TODO
+```
+
+> The above command returns JSON document:
+
+```json
+{
+    "version": "1",
+    "success": true,
+    "data": {
+        "last_validated_child_block_number": 10000,
+        "last_mined_child_block_timestamp": 1535031020,
+        "last_mined_child_block_number": 11000,
+        "eth_syncing": true,
+        "byzantine_events":
+        [
+            {
+                "event": "invalid_exit",
```

Comment:
ok, so moving this approach forward, we should have a similar construction for other conditions? Like (following status quo here, we can structure differently, if you think it's better)
`:invalid_block`, `:block_withholding`, `:unchallenged_exit`, `:invalid_piggyback`, `submit_competitor` (`invalid_ife`? `invalid_canonicality`? split to `:canonical_ife` + `:noncanonical_ife`, where the title would reflect what's byzantine?), `:piggyback_required` (`:piggyback_possible` depending on how we serve the user) , (optional) `tx_requires_ife`

(I'm making the names up now, based on the MoreVP diagram, sorry for any inconsistency).

CC @purbanow

Reply:
Right.
I'm not going to add any of these now though - the docs should be updated as these features are implemented

Reply:
Yes, LGTM, but let's keep challenging this design and refer to MoreVP specs to be sure there aren't simpler/cleaner alternatives. To not fall in the trap of overspeccing (I tend to do that a lot ;) ).

When we settle on on that, I can update the [diagram](https://docs.google.com/drawings/d/1UaAMZTJBbikTM0eFSbNM7ZVgi1HZrLTfwYCUvHGUaW0/edit) to be more in sync with the names and division. But for now let's keep as is, conceptually it is the same.

With `noncanonical_ife` and `invalid_ife_challenge` being the same event, I was hoping they could be merged, but now, [here](https://github.com/omisego/plasma-contracts/blob/elixir_omg_morevp_compat/contracts/RootChain.sol#L526) and [here](https://github.com/omisego/plasma-contracts/blob/elixir_omg_morevp_compat/contracts/RootChain.sol#L579) I see there are differences in ways to respond are big enough to separate.

Reply:
There's a fundamental question though, that I keep bumping into, so I'll drop it here.

Separation/merging between "telling an ACTION is required" and "giving all info to perform it".

1/ Currently we're doing the latter - push event (or `status/`) returns a "reference" to an ACTION (`:invalid_exit` with a respective utxo position to challenge) and then the user asks for details (proofs etc) on another endpoint `/utxo.get_challenge_data utxo_pos`
2/ there are performance issues - getting the full information is likely to be heavier than just the prompt, especially when we go in the "security critical Watcher" mode
3/ OTOH, API-wise, the thinnest API is when you just drop everything on the user in `status/` and not have `/utxo.get_challenge_data utxo_pos`
4/ OTOH^2, to be consistent, we'd need to move the `/utxo.get_challenge_data utxo_pos` which is additional cleanup work

So 2 vs 3 vs 4 needs to be considered. I'm slightly in favor of not removing `/utxo.get_challenge_data utxo_pos` and adding its MoreVP counterparts, but I might be wrong.

Reply:
I lean towards separating "reporting of the state" from "giving all info necessary to respond to it".
I think that there will be multiple polls to `/status`, but the user will only need to get the necessary info to perform the action once, so including all the info each time seems unnecessarily heavyweight.